subsys/bluetooth/shell/bt.c

```diff
@@ -1147,14 +1160,83 @@ static int cmd_scan_filter_clear_addr(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_OBSERVER */
 
 #if defined(CONFIG_BT_BROADCASTER)
-static const struct bt_data ad_discov[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-};
+static void ad_mode_parse(const struct shell *sh, size_t argc, char *argv[], bool *discov)
+{
+	for (size_t argn = 0; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "discov")) {
+			*discov = true;
+			break;
+		} else if (!strcmp(arg, "non_discov")) {
+			*discov = false;
+			break;
+		}
+	}
+}
+
+static ssize_t ad_init(struct bt_data *data, const size_t data_size, const atomic_t *adv_opt)
+{
+	const bool discoverable = atomic_test_bit(adv_opt, SHELL_ADV_OPT_DISCOVERABLE);
+	const bool adv_ext = atomic_test_bit(adv_opt, SHELL_ADV_OPT_EXT_ADV);
+	static uint8_t ad_flags;
+	size_t ad_len = 0;
+
+	/* Set BR/EDR Not Supported if LE-only device */
+	ad_flags = IS_ENABLED(CONFIG_BT_BREDR) ? 0 : BT_LE_AD_NO_BREDR;
+
+	if (discoverable) {
+		/* A privacy-enabled Set Member should advertise RSI values only when in
+		 * the GAP Limited Discoverable mode.
+		 */
+		if (IS_ENABLED(CONFIG_BT_PRIVACY) && IS_ENABLED(CONFIG_BT_CSIS) && csis != NULL) {
+			ad_flags |= BT_LE_AD_LIMITED;
+		} else {
+			ad_flags |= BT_LE_AD_GENERAL;
+		}
+	}
+
+	if (ad_flags != 0) {
+		__ASSERT(data_size > ad_len, "No space for AD_FLAGS");
+		data[ad_len].type = BT_DATA_FLAGS;
+		data[ad_len].data_len = sizeof(ad_flags);
+		data[ad_len].data = &ad_flags;
+		ad_len++;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_CSIS)) {
```

Comment:
this part seems like it should rather be in csis.c or audio.c, same for the externs in bt.h

Reply:
We could move the CSIS stuff into `audio_ad_data_add`. 

CSIS can be considered outside the scope of audio (As it can used for non-audio stuff), so maybe it is better to put into the csis.c shell module instead.

Reply:
Yeah I have a bunch to catch up on in the spec. I hope I'll have some time to do that once we've (aleks and I) addressed the more critical host bugs.

Reply:
So, as I understand, It can be left as it is, since `csis_ad_data_add`called here, is in `csis.c` already

Reply:
Indeed